R/pcairPartition.R

```diff
@@ -9,40 +9,64 @@ pcairPartition <- function(kinobj, divobj,
 
     # sample.id from kinship matrix
     kin.id.all <- .readSampleId(kinobj)
-    # sample.id from divergence matrix
-    div.id.all <- .readSampleId(divobj)
+    id.all <- kin.id.all
 
     # check for ids provided
-    if(is.null(kin.id.all) | is.null(div.id.all)) {
-        stop('colnames must be provided for kinobj and divobj')
-    }
-
-    # logical indicators of which samples are in both sets
-    kin.read <- kin.id.all %in% div.id.all
-    div.read <- div.id.all %in% kin.id.all
-    if(!(all(kin.read) & all(div.read))){
-        warning('kinobj and divobj contain non-matching samples; only partitioning those present in both objects')
+    if(is.null(kin.id.all)) {
+        stop('colnames must be provided for kinobj')
     }
 
     # filter to samples in sample.include
+    kin.read <- rep(TRUE, length(kin.id.all))
```

Comment:
@smgogarten Not that it really matters, but I don't think we need this line unless the next if is false, so it could be an else statement.

Reply:
Agreed, it's more readable that way.